### PR TITLE
Adds torch.linalg namespace

### PR DESCRIPTION
**Pull request by mruberry**
This PR adds the `torch.linalg` namespace as part of our continued effort to be more compatible with NumPy. The namespace is tested by adding a single function, `torch.linalg.outer`, and testing it in a new test suite, test_linalg.py. It follows the same pattern that https://github.com/pytorch/pytorch/pull/41911, which added the `torch.fft` namespace, did. 

Future PRs will likely:

- add more functions to torch.linalg
- expand the testing done in test_linalg.py, including legacy functions, like torch.ger
- deprecate existing linalg functions outside of `torch.linalg` in preference to the new namespace

### CODE REVIEW

**Comment by mattip** [comment]
[Here](https://numpy.org/devdocs/reference/routines.linalg.html?highlight=linalg#module-numpy.linalg) is the reference page for numpy.linalg. The equivalent pytorch functions:

done? | NumPy (`numpy.linalg`) | PyTorch (`torch.*` unless otherwise noted)| notes (differences in arguments)
--- | --- | --- | ---
. | dot | dot | pytorch does not broadcast, does not have an `out` argument
. | multi-dot | X | does not exist yet
. | vdot | X | If the first argument is complex the complex conjugate of the first argument is used for the calculation of the dot product.
. | inner | dot? | but there are special cases ...
yes | outer | ger | 
. | matmul | matmul 
. | tensordot | tensordot 
. | einsum | einsum  
. | matrix_power | | Raise a square matrix to the (integer) power n. No stacked matrices
. | kron | |
. | cholesky | cholesky
. | qr | qr | interface for mode is different
. | svd | svd | interface for mode is different
. | eig | eig  | interface to return both eigenvalues and eigenvectors is different
. | eigh | X
. | eigvals | X
. | eigvalsh | X 
. | norm | norm 
. | cond | X
. | det | det
. | matrix_rank | matrix_rank
. | slog_det | slogdet
. | trace | trace | interface is different
. | solve | solve
. | tensorsolve | X | numpy has an axes arg beyond `solve`
. | lstsq | lstsq | numpy has an extra rcond arg
. | inv | inverse | pytorch adds an `out` arg
. | pinv | pinverse | numpy has an extra `hermatian` boolean arg
. | tensorinv | X | inverse of tensordot

Should I move this to a tracking issue?

**Comment by mruberry** [comment]
@mattip A tracking issue with those functions would be great!

File: torch/linalg/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import sys
+
+import torch
+from torch._C import _add_docstr, _linalg  # type: ignore
+
+Tensor = torch.Tensor
+
+# Note: This not only adds the doc strings for the spectral ops, but
+# connects the torch.fft Python namespace to the torch._C._fft builtins.
```
Comment (kurtamohler) [inline torch/linalg/__init__.py:9]:
```suggestion
# connects the torch.linalg Python namespace to the torch._C._linalg builtins.
```

**Review by gchanan — APPROVED** [review]

File: torch/linalg/__init__.py (hunk 1 above)
Comment (mruberry) [reply torch/linalg/__init__.py:9]: Great catch! Thank you, @kurtamohler!

**Pull request closed without merging**